nntrainer/include/weight.h

```diff
@@ -101,6 +101,20 @@ class Weight {
     swap(lhs.name, rhs.name);
   }
 
+  /**
+   * @brief Copy constructor for weight
+   *
+   * @param rhs weight to construct from
+   */
+  Weight(const Weight &rhs) = default;
```

Comment:
is this essentially needed? If some layer writer uses these, then the weights of both the layers would be bonded together
and create nasty bugs.

Reply:
If we have a copy assignment, we need a copy constructor.  If we are gonna delete a copy assignment, I think it's fine for me.  I am gonna be using this for the integrated tests, btw, but I can work around I think we decide to not allow copy ctor/assignment.

I got what you mean though, it seems that there should be a broader patch that changes copy assignment as well.

Reply:
Copy assignment is wrong. 
It is currently only used in `Layer::copy` with wrong usage. It must create a deep copy of the weights but it rather binds the weights of the layer together as copy constructor of Tensor binds the memory together.